src/main/java/net/openhft/chronicle/core/io/DualReferenceCounted.java

```diff
@@ -58,8 +59,11 @@ private void checkError() {
             throw new AssertionError("Unable to use this resource due to previous error", error);
         int aRefCount = a.refCount();
         int bRefCount = b.refCount();
-        if (aRefCount != bRefCount)
-            throw Jvm.rethrow(error = new AssertionError(aRefCount + " != " + bRefCount, error));
+        if (aRefCount != bRefCount) {
+            final AssertionError ae = new AssertionError(aRefCount + " != " + bRefCount, error);
+            error = ae;
```

Comment:
Why not just error = new AssertionError; throw (error)? What's `ea` for?

Reply:
That's how I started but got this:

![image](https://user-images.githubusercontent.com/7457876/146385713-944ed4ca-dcef-497b-a8db-862d7b4a21e2.png)

Reply:
Perhaps a false positive?

Reply:
This does make sense.